src/GraphQL/Validation/TypeInfo.cs

```diff
@@ -274,5 +280,39 @@ private FieldType GetFieldDef(ISchema schema, IGraphType parentType, Field field
 
             return null;
         }
+
+        /// <summary>
+        /// Tracks already visited fragments to maintain O(N) and to ensure that cycles
+        /// are not redundantly reported.
+        /// </summary>
+        internal HashSet<string> NoFragmentCycles_VisitedFrags;
+        /// <summary>
+        /// Array of AST nodes used to produce meaningful errors
+        /// </summary>
+        internal Stack<FragmentSpread> NoFragmentCycles_SpreadPath;
+        /// <summary>
+        /// Position in the spread path
+        /// </summary>
+        internal Dictionary<string, int> NoFragmentCycles_SpreadPathIndexByName;
+
+        internal HashSet<string> NoUndefinedVariables_VariableNameDefined;
+
+        internal List<Operation> NoUnusedFragments_OperationDefs;
+        internal List<FragmentDefinition> NoUnusedFragments_FragmentDefs;
+
+        internal List<VariableDefinition> NoUnusedVariables_VariableDefs;
+
+        internal Dictionary<string, Argument> UniqueArgumentNames_KnownArgs;
+
+        internal Dictionary<string, FragmentDefinition> UniqueFragmentNames_KnownFragments;
+
+        internal Stack<Dictionary<string, IValue>> UniqueInputFieldNames_KnownNameStack;
+        internal Dictionary<string, IValue> UniqueInputFieldNames_KnownNames;
+
+        internal HashSet<string> UniqueOperationNames_Frequency;
+
+        internal Dictionary<string, VariableDefinition> UniqueVariableNames_KnownVariables;
+
+        internal Dictionary<string, VariableDefinition> VariablesInAllowedPosition_VarDefMap;
```

Comment:
In order to realize 100% static lambdas inside validation rules, these members were added to `TypeInfo`.  Unfortunately they use memory even if the rule is not in use.  However, they do not use additional allocations since they remain set to `null`.  This is another 112 bytes on a 64-bit computer.  I'm not sure, but this could be what's making there be an increased number of gen 1 collections - probably the size of the class now pushes it into a gen 1 heap from the start.  If that's the reason, I could probably try moving these members into another class...

Note that doing this represents a very small improvement over declaring the variable inside the validation rule and allowing the rule to allocate memory for the lambda.  So I do not think it worth our time to figure out how to provide this benefit to third-party validation rules.  It's just too miniscule of a improvement.